### DISCUSSION TITLE
Deployment Procedure Clarification

### DISCUSSION THREAD

**Discussion by ujjyal** [Accenture/sfmc-devtools#1566] [Q&A] [answered]
HI,

Previously i was using the following sequence of steps for my market deployment

-  retrieve as template (rt)
- Build as template (bt)
- Build Definition (bd)
- Deploy (d)

But while running this in the new version of the tool. I am getting error. skipping query XXXX : Dependent dataExtension with CustomerKey='undefined' was not found on your BU. With the new set up, the query activities that performs actions on shared data extension, are not working any more.

So I thought it is a issue because of "retrieve as template (rt)", as this is deprecated. But then reading the wiki [https://github.com/Accenture/sfmc-devtools/wiki/06.c-~-Templating-Commands](url). It is not very clear for me the sequence of commands that I have to use.
I have tried with

- Build 
- deploy
but this gives me error <Please create a directory called deploy and include your metadata in it>. 
But since build command includes both **bt** and **bd** and **retrieve** command all, I thought the manual creation of the folder is not necessary, as it was previously taken care by the earlier sequence of commands

Can you please suggest a way ahead.

Thanks,
Ujjyal

**Comment by ujjyal** [comment, marked as answer]
I Have solved the issue. I have to perform retrieve first.

**Reply by JoernBerkefeld** [reply]
Hi @ujjyal - you indeed need to combine build and deploy. The build command will offer retrieving relevant files for you in the latest version of mcdev. Did you say "no" to that offer before?

I can also strongly recommend to add '--dependencies' to the build command as that ensures you get all dependent files into your deploy folder as well.

I will update the templating docs in the coming days to showcase a normal flow as I think you right about that page lacking detail.
